### PR TITLE
3678 solution capabilities validation

### PR DESCRIPTION
**Pull request by RuthBerry96**

### CODE REVIEW

File: src/NHSD.BuyingCatalogue.Solutions.Contracts/Persistence/ISolutionCapabilityRepository.cs [hunk 1]
```diff
@@ -9,5 +9,8 @@ public interface ISolutionCapabilityRepository
         Task<IEnumerable<ISolutionCapabilityListResult>> ListSolutionCapabilities(string solutionId, CancellationToken cancellationToken);
 
         Task UpdateCapabilitiesAsync(IUpdateCapabilityRequest updateCapabilityRequest, CancellationToken cancellationToken);
+
+        Task<int> GetMatchingCapabilitiesCount(IEnumerable<string> capabilitiesToMatch,
```
Comment (sam-roberts-bjss) [inline src/NHSD.BuyingCatalogue.Solutions.Contracts/Persistence/ISolutionCapabilityRepository.cs]: Async?

File: src/NHSD.BuyingCatalogue.Solutions.Persistence/Repositories/SolutionCapabilityRepository.cs [hunk 2]
```diff
@@ -26,6 +27,13 @@ FROM SolutionCapability
                                 WHERE SolutionCapability.SolutionId = @solutionId
                                 ORDER BY Capability.Name";
 
+        private const string checkCapabilitiesExist = @"SELECT COUNT(*)
+                                                        FROM
+                                                       (SELECT CapabilityRef
+                                                        FROM Capability
+                                                        WHERE CapabilityRef in @capabilitiesToMatch
+                                                        GROUP BY CapabilityRef) AS a";
```
Comment (sam-roberts-bjss) [inline src/NHSD.BuyingCatalogue.Solutions.Persistence/Repositories/SolutionCapabilityRepository.cs]: As count?

File: tests/NHSD.BuyingCatalogue.API.IntegrationTests/Features/Capability/Update Capabilities Validation.3678.feature [hunk 3]
```diff
@@ -0,0 +1,45 @@
+Feature: Update Capabilities Validation
+	As a Public User
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.API.IntegrationTests/Features/Capability/Update Capabilities Validation.3678.feature]: Indenting?

File: tests/NHSD.BuyingCatalogue.API.IntegrationTests/Features/Capability/Update Capabilities Validation.3678.feature [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	And Suppliers exist
+		| Id    | SupplierName |
+		| Sup 1 | Supplier 1   |
+	And Solutions exist
+		| SolutionID | SolutionName   | SupplierStatusId | SupplierId |
+		| Sln1       | MedicOnline    | 1                | Sup 1      |
+		| Sln2       | TakeTheRedPill | 1                | Sup 1      |
+	And Solutions are linked to Capabilities
+		| Solution       | Capability          |
+		| TakeTheRedPill | Resource Management |
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.API.IntegrationTests/Features/Capability/Update Capabilities Validation.3678.feature]: Should this be using the SolutionId version same as Scenario 1?

File: tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/Solutions/UpdateSolutionCapabilitiesTests.cs [hunk 5]
```diff
@@ -32,6 +32,10 @@ public async Task ShouldUpdateCapabilitiesAsync()
         {
             var capabilityRefs = new HashSet<string>(){"C1", "C2"};
 
+            _context.MockSolutionCapabilityRepository.Setup(c =>
+                    c.GetMatchingCapabilitiesCountAsync(capabilityRefs, It.IsAny<CancellationToken>()))
+                .ReturnsAsync(2);
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/Solutions/UpdateSolutionCapabilitiesTests.cs]: Could we constant the 2 to give a little meaning to it e.g. expectedCapabilityCount

File: tests/NHSD.BuyingCatalogue.API.IntegrationTests/Features/Capability/Update Capabilities Validation.3678.feature (hunk 4 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.API.IntegrationTests/Features/Capability/Update Capabilities Validation.3678.feature]: This is the seperate task that needs to be done. #4894

**Pull request merged**